code/datums/components/squeak.dm

```diff
@@ -40,10 +39,10 @@
 
 /datum/component/squeak/proc/play_squeak()
 	if(prob(squeak_chance))
-		if(!override_squeak_sounds)
-			playsound(parent, pickweight(default_squeak_sounds), volume, 1, -1)
+		if(!override_squeak_datum)
+			playsound(parent, /datum/outputs/bikehorn, volume, 1, -1)
```

Comment:
here if the var wasnt set right for some reason you send a pure type while the code below is made to ignore pure types
if there isnt a datum in a var then just throw an error that for some reason var was deleted from the atom